File: selection_sort.py
' selection sort in python '

def swap(A,x,y):
  tmp = A[x]
  A[x] = A[y]
  A[y| = tmp

def selectionSort(myList):
  for i in range(len(myList)):
    least = i
    
    for k in range(i+1,len(myList)):
      if myList[k] < myList[least]:
        least = k
        
      swap(myList,least,i)

File: selection_sort.js
/* selection sort in javascript */

function swap (a,b)
{
  var tmp;
  tmp = a;
  a = b;
  b = tmp;
}

function selectionSort (anArray)
{
  var i, j, tmp;
  for (i=0; i < anArray.length-1; i++)
  {
    tmp = i;
    for (j=i+1; j < anArray.length; j++)
    {
      if (anArray[j] < anArray[tmp]) {
        tmp = j;
      }
    }
    var a, b = swap(anArray[i], anArray[j])
    }
  }
}
